src/firmware/differential_drive_controller/differential_drive_controller.ino

```diff
@@ -80,41 +123,60 @@ void setup() {
 }
 
 void loop() {
-
   // read RC input
-  rc_read();
+  #ifndef NO_RC
+    rc_read();
+  #endif
   
+  // If we're debugging serial, we always want to be reading and printing out 
+  // the values received over serial, even if we don't use them 
+  #ifdef DEBUG_SERIAL
+    if (Mode != 1)
+      serial_read();
+  #endif
   
-  if (Mode == 1) { // Auto Mode
+  if (Mode == 1) { // Autonomous Mode
     serial_read(); 
     
     convert();
     drive(linear_x, angular_z);
   }
-  else if (Mode == 0) { //RC Mode
+  else if (Mode == 0) { // RC Mode
     linear_x = range1; 
     angular_z = range2;
+    // TODO: Set other values to 0 here
     
     convert();
     drive(linear_x, angular_z);
-    Serial.flushRX();
   }
-  else { // STOP MODE
+  else { // E-STOP MODE
     linear_x = UNMAPPED_STOP_SPEED; 
     angular_z = UNMAPPED_STOP_SPEED;
+    // TODO: Set other values to 0 here
     
     convert();
     drive(linear_x, angular_z);
-    Serial.flushRX();
   }
+  
+  #ifdef DEBUG_COMMANDS
+    Serial.println("Command going to wheels:");
+    Serial.print("Linear X: ");Serial.println(linear_x);
+    Serial.print("Linear Y: "); Serial.println(linear_y);
+    Serial.print("Linear Z: ");Serial.println(linear_z);
+    Serial.print("Angular X: ");Serial.println(angular_x);
+    Serial.print("Angular Y: ");Serial.println(angular_y);
+    Serial.print("Angular Z: ");Serial.println(angular_z);
+    Serial.println();
+  #endif
 }
 
 /*
 * Calculates OFFSET for the joystick controllers. 
 */
```

Comment:
Include in operating procedure that the reset button should be pressed at the beginning of runs with the following conditions: 
1. E-stop down
2. Remote on in neutral throttle on both joysticks